server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/ToSenderFolder.java

```diff
@@ -71,12 +71,10 @@ public ToSenderFolder(UsersRepository usersRepository, @Named("mailboxmanager")
      */
     @Override
     public void service(Mail mail) throws MessagingException {
-        if (!mail.getState().equals(Mail.GHOST)) {
```

Comment:
With cassandra distributed version: I'd like to save the sent letter to Sent folder. When I add the ToSenderFolder to the mailetcontainer(as shown in the commit), it doesn't work correctlly, the letter couldn't be saved in the Sent folder because of the mail state becomes to Ghost.
I think if someone want's to save the letter to Sent,  we shouldn't care the mail's sate, no matter what sate it is, only if  the ToSendFolder.doService methods is called, it should work. I'm not sure what I think is correct, what's your opion? @chibenwa

Reply:
Or do you have any other better solution to solve the issue?
1. save the letter to sent
2. If  "to" is local domain, "cc" is the other domain, it saves two copies of the letter to Sent folder(which is unacceptable）

Reply:
Have you been configuring LocalDelivery with `<consume>false</consume>` ?

Reply:
Also there should be only one email transiting, with "both to and cc positioned", and the envoleope should cary out both To and Cc recipients, ensuring the email is saved only once.

I don't see how your changeset would be chaging anything.

Can you write a test?

Reply:
**No answer here**